internal/resolution/variablesources/bundle.go

```diff
@@ -2,62 +2,36 @@ package variablesources
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
-	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
-var _ input.VariableSource = &BundlesAndDepsVariableSource{}
-
-type BundlesAndDepsVariableSource struct {
-	catalogClient   BundleProvider
-	variableSources []input.VariableSource
-}
-
-func NewBundlesAndDepsVariableSource(catalogClient BundleProvider, inputVariableSources ...input.VariableSource) *BundlesAndDepsVariableSource {
-	return &BundlesAndDepsVariableSource{
-		catalogClient:   catalogClient,
-		variableSources: inputVariableSources,
+func MakeBundleVariables(
+	allBundles []*catalogmetadata.Bundle,
+	requiredPackages []*olmvariables.RequiredPackageVariable,
+	installedPackages []*olmvariables.InstalledPackageVariable,
+) ([]*olmvariables.BundleVariable, error) {
+	bundleQueue := make([]*catalogmetadata.Bundle, 0, len(requiredPackages)+len(installedPackages))
+	for _, variable := range requiredPackages {
+		bundleQueue = append(bundleQueue, variable.Bundles()...)
 	}
-}
-
-func (b *BundlesAndDepsVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
```

Comment:
A lot of this diff seems like movement of code to later in the file - can we undo that to make the diff as minimal as we can? Hard to tell if you changed the code in `GetVariables()` or not.

Reply:
I'm not sure that I can minimise the diff further. I took out the part out of `GetVariables()` and moved into `MakeBundleVariables`. Also I'm happy that the important bits (where we generate variables) is now in `MakeBundleVariables` and history of these lines will be preserved.

`BundlesAndDepsVariableSource` and its `GetVariables` is now just a bridge between old structure and new structure. It will be completely removed in #501.

Reply:
I think you might be able to - try moving the code at the bottom of the file to the top and see.

Reply:
You are right. It produced a smaller diff and still preserves history for `MakeBundleVariables`. I'll push an update shortly.